modules/apps/journal/journal-content-web/src/main/java/com/liferay/journal/content/web/internal/portlet/JournalContentPortlet.java

```diff
@@ -126,19 +130,46 @@ public void doView(
 				_log.error("Unable to get journal article", portalException);
 			}
 		}
-		else if ((articleGroupId > 0) && Validator.isNotNull(articleId)) {
+		else if ((articleGroupId > 0) &&
+				 (Validator.isNotNull(articleExternalReferenceCode) ||
+				  Validator.isNotNull(articleId))) {
```

Comment:
I think this condition is wrong. Isn't it possible for articleId != null, articleERC == null and FF.isEnabled(LPD-27566) == true? Wouldn't that fail with a NPE?

I think the condition has to be something like:
```
(Validator.isNotNull(ERC) && FF.isEnabled(...)) || (Validator.isNotNull(articleId) && !FF.isEnabled(...))
```